editor.planx.uk/src/@planx/components/shared/Schema/Fields.tsx

```diff
@@ -178,11 +187,10 @@ export const SelectFieldInput: React.FC<Props<QuestionField>> = (props) => {
 };
 
 export const ChecklistFieldInput: React.FC<Props<ChecklistField>> = (props) => {
-  const { formik, activeIndex } = useListContext();
-  const {
-    id,
-    data: { options, title, fn },
-  } = props;
+  const { data: { title, options }, formik } = props;
+  const { id, errorMessage, name, value } = getFieldProps(props);
+
+  if (!Array.isArray(value)) throw Error("'value' prop for ChecklistFieldInput must be of type string[]");
```

Comment:
Error handling is required here for type narrowing as `SchemaResponse` has no relationship to a `Field. It's not currently able to know that `value` will be a `string[]` here as it's derived from a `ChecklistFiled`.

We also see this in lines of code like this that require a manual type narrowing - 

https://github.com/theopensystemslab/planx-new/blob/main/editor.planx.uk/src/@planx/components/List/utils.tsx#L61-L63

This would be more meaningful / maintainable if the logic was something like - 

```
if (!field.type === "Checklist") {
  sum += parseInt(item?.identicalUnits);
}
```

As the range of fields grows (and the options in `SchemaResponse) this feels like it will be an import consideration.

There's a few solutions for this, I'll open a follow up PR shortly.